### PR TITLE
Switch from Azure to GitHub Actions

### PR DESCRIPTION
**Pull request by astrofrog**
I am trying out the new re-usable OpenAstronomy workflows. For now, I am using @ConorMacBride's fork for the tox testing. This also now adds a publish workflow.

Fixes https://github.com/astropy/extension-helpers/issues/8

### CODE REVIEW

**Comment by ConorMacBride** [comment]
The tox workflow is now merged so you should be able to use the OpenAstronomy main branch.

**Comment by astrofrog** [comment]
@ConorMacBride - just curious, do you have any idea what could be happening in the Windows build here? (it's complaining about not finding the Visual C++ compiler which is needed to build a package with a C extension in the test - I'm curious why this wasn't an issue on Azure though)

**Comment by ConorMacBride** [comment]
It's saying "Microsoft Visual C++ 14.0 or greater is required" however version should be install in the windows-latest image: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#microsoft-visual-c

Unless an environment variable is missing (or the test is run isolated from the default environment variables)?

**Comment by astrofrog** [comment]
Yeah I was wondering if it was a tox isolation thing - it's just curious because this didn't occur on Azure. I'll try and dig deeper.

**Comment by ConorMacBride** [comment]
Yeah, maybe disabling the tox isolation will make it work? Azure Pipelines is using the windows-2019 image (https://github.com/actions/virtual-environments/blob/win19/20220131.1/images/win/Windows2019-Readme.md) while GitHub Actions is using windows-2022. The windows job seems to expect OpenMP to be installed, although nether of these images lists that as being installed so how did it work before?

**Comment by saimn** [comment]
I think there was an issue with the recent Windows upgrade on Actions and the Visual C++ compiler. @pllim probably knows more ?

**Comment by pllim** [comment]
Yes, @lpsinger fixed it by downgrading to windows-2019

**Comment by ConorMacBride** [comment]
Should we temporarily downgrade Windows in the reusable workflows from `windows-latest` to `windows-2019`? Would it be worth adding a way to specify a custom image?

**Comment by lpsinger** [comment]
> Yes, @lpsinger fixed it by downgrading to windows-2019

See https://github.com/actions/virtual-environments/issues/5141

**Comment by pllim** [comment]
You only have to downgrade if you are compiling C, so 🤷

**Comment by astrofrog** [comment]
@ConorMacBride - I think it's probably best to downgrade to windows-2019 for now, it's not like we really need the more recent version?

**Comment by astrofrog** [comment]
Ah I just saw https://github.com/OpenAstronomy/github-actions-workflows/pull/15

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        - macos: py38-test-dev
+          posargs: --openmp-expected=False
+        - windows: py36-test
+        - windows: py38-test-dev
+        # Test with more compilers, for the OpenMP helpers
+        - macos: py38-test-osxclang-conda
+        - linux: py38-test-linuxgcc-conda
+        # Test downstream packages
+        - linux: py39-downstream
+
```
Comment (ConorMacBride) [inline .github/workflows/main.yml:31]: This should make it use the windows-2019 image on Windows:
```suggestion
    env:
      OA_WINDOWS_RUNNER: windows-2019

```

Comment (astrofrog) [reply .github/workflows/main.yml]: Oops just saw this!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          posargs: --openmp-expected=False
+        - windows: py36-test
+        - windows: py38-test-dev
+        # Test with more compilers, for the OpenMP helpers
+        - macos: py38-test-osxclang-conda
+        - linux: py38-test-linuxgcc-conda
+        # Test downstream packages
+        - linux: py39-downstream
+      env:
+        OA_WINDOWS_RUNNER: windows-2019
```
Comment (astrofrog) [inline .github/workflows/main.yml:31]: @ConorMacBride - I can't figure out what the correct level of indentation needs to be, nothing seems to work. Do you have any ideas?

Comment (pllim) [reply .github/workflows/main.yml:31]: Doesn't the web editor tell you if you use GitHub web UI to edit the file? You indent/unindent until the red swiggles go away... 😆

Comment (ConorMacBride) [reply .github/workflows/main.yml:31]: Apparently you can't pass custom environment variables to a reusable workflow: https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations

I'm working on a new `runner` input.

Comment (ConorMacBride) [reply .github/workflows/main.yml:31]: You can now remove the `OA_WINDOWS_RUNNER` and do:
```yaml
- windows: py36-test
  runs-on: windows-2019
- windows: py38-test-dev
  runs-on: windows-2019
```

Comment (astrofrog) [reply .github/workflows/main.yml]: Thanks! I've added that here.

**Comment by ConorMacBride** [comment]
The pinned `jinja2==2.10.3` in tox.ini is trying to import a name in a dependency which has now been removed. Is this pin needed? jinja2 is now used by pytest-mpl to produce HTML reports and sunpy is using pytest-mpl, although just running `pytest --pyargs sunpy` doesn't enable pytest-mpl, it still gets imported. Maybe the jinja import in pytest-mpl should be made made lazy?

**Comment by astrofrog** [comment]
@Cadair - just FYI I've set PYPI_TOKEN in the repo secrets

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+on:
```
Comment (Cadair) [inline .github/workflows/main.yml:1]: Can you name workflows? Make the checks look nicer?

Comment (astrofrog) [reply .github/workflows/main.yml:3]: Done, I just went with CI to be concise

File: setup.cfg [hunk 4]
```diff
@@ -36,6 +36,7 @@ setuptools.finalize_distribution_options =
 
 [options.extras_require]
 test =
+    wheel
```
Comment (Cadair) [inline setup.cfg:39]: Why?

**Review by Cadair — APPROVED** [review]
I like it, the templates look good.

File: setup.cfg (hunk 4 above)
Comment (astrofrog) [reply setup.cfg:39]: The test C extension package failed otherwise for some reason.

Comment (Cadair) [reply setup.cfg:39]: curious

**Comment by Cadair** [comment]
We waiting for the first release to merge this?

**Comment by astrofrog** [comment]
@Cadair - I've updated the ref for the workflows to v1, feel free to merge if the CI passes!

**Pull request merged**